runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -112,11 +127,6 @@ GC_ArrayletObjectModel::getArrayletLayout(J9Class* clazz, UDATA dataSizeInBytes,
 			if (extensions->isVLHGC()) {
 				adjustedHybridSpineBytesAfterMove += objectAlignmentInBytes;
 			}
-#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
-			if (extensions->indexableObjectModel.isDoubleMappingEnabled()) {
-				layout = Discontiguous;
```

Comment:
seems like you are changing behaviour for case when isVirtualLargeObjectHeapEnabled is false but isDoubleMappingEnabled - layout was Discontiguous and it's will be InlineContiguous

can't tell yet (without fully understand other spots like jni critical how you deal with it), but ATM I feel you should not change that behavior.....

bottom line, since I don't know if we have any testing for that combo on regular basis,  you should locally test it. obviously the workload also has to create discontigous arrays

Reply:
I forgot some details, but seems like for that combination you assume dataAddr is present, and now double mapping will populate and use dataAddr the same way as virtualLargeObjectHeap. So, you unify the logic for 2 complementary combinations (one being virtualLargeObjectHeap enabled, and doubleMapping disabled and the other one being virtualLargeObjectHeap disabled and doubleMapping enabled) - it's just the mechanism how we get the address for off-heap data is different. Did I get it right?

While historically dataAddr presence was not required for double mapping, the assumption that it will be present from now on is suboptimal for footprint, but still might be ok, if the code gets simpler. 

Eventually, if virtualLargeObjectHeap  works well, we will probably remove double mapping support anyway...

Btw, how do we deal with situations if someone tries invalid combinations, like:
- both virtualLargeObjectHeap and doubleMapping enabled
- virtualLargeObjectHeap disabled, doubleMapping enabled, but dataAddr not present
  (I suspect JIT will not be flexible enough to run with Balanced GC and dataAddr not present, regardless if doubleMapping is enabled or not)

Reply:
>Did I get it right?
Yes 👍 

>Eventually, if virtualLargeObjectHeap works well, we will probably remove double mapping support anyway...
👍 

>If both virtualLargeObjectHeap and doubleMapping enabled?
There is no warning message and virtualLargeObjectHeap will take precedence.

>If virtualLargeObjectHeap is disabled, doubleMapping is enabled, but dataAddr is not present?
This scenario would not be possible. If double mapping is enabled, dataAddr will be present. Gencon and other policies that may not have dataAddr will run as they normally would. Adding `-XXgc:enableVirtualLargeObjectHeap` or `-Xgc:enableArrayletDoubleMapping` will only request these options to be enabled, however, only in balanced do we have a pass to actually process these requests and enable the functionality.